### DISCUSSION TITLE
[NomadNet] What is an Identity?

### DISCUSSION THREAD

**Discussion by LinuxinaBit** [markqvist/Reticulum#409] [Help & Questions] [answered]
I know what a LXMF address is, but what is the Identity below it?

<img width="423" alt="image" src="https://github.com/markqvist/Reticulum/assets/105567407/705effd7-be9d-455d-8a26-5ec23e775a84">

**Comment by faragher** [comment, marked as answer]
An identity is the core of a Reticulum user. (FYI: that hash is safe to publish, but the file itself contains your keys, so protect that)

The identity contains the private key and is the seed by which all other addresses are derived. Your LXMF address is deterministically derived from your identity and lxmf.delivery - no matter how many times you derive the address, or who derives it, you get the same address. If someone knows your LXMF address they can send you a message. If they know your identity, they can find the address to any destination you create, so long as they know the app name and aspects. This it not a security risk, since if those destinations aren't announced, no traffic will be routed.

As a receiver this is important for keeping your addresses deterministic and sane. As a sender, this means you can find other addresses dynamically. For example, sensor.temperature and sensor.humidity

At a broader scale, it also means you can do things like weather.london and weather.kiev and have them served by different servers in different nations, so long as they're all the same identity. It opens up a huge number of options and prevents the need for a proxy to route traffic at the endpoint. Most of this is beyond the typical use-case, but it's a fascinating path to the future.

But apart from the why, the what is as I said, it's an identity that defines you as a person or organization and contains your cryptographic keys. Using a new one on each machine is the typical use case.

**Reply by LinuxinaBit** [reply]
Thank you for the great answer.
I was a bit worried because I did publish the hash, thank you for that clarification as well :)

**Reply by trosel** [reply]
> Using a new one on each machine is the typical use case.

you mean to say that a user would have a new LXMF address on each machine, but they would all have the same Identity, right?

**Reply by faragher** [reply]
No, an LXMF address is determined by the identity and only one address will receive a given transmission. If you generate a new lxmf.delivery address on a known identity, you'll end up with the same address again. Generally speaking it's one identity per machine, Having a multiple access identity is not really supported by LXMF.

For a more E-Mail like experience, you'd need a dedicated mailbox application that would, ideally, have a list of authorized identities that would identify themselves and use that for authentication. LXMF is not as feature rich as POP3 or SMTP, and a real mail protocol would be very helpful.
